lib/codecs/src/internal_events/encoder.rs

```diff
@@ -0,0 +1,58 @@
+use metrics::counter;
+use tracing::error;
+use vector_core::internal_event::InternalEvent;
+
+#[derive(Debug)]
+pub struct GelfSerializeFailedMissingField<'a> {
+    pub message: &'a str,
+    pub field: &'a str,
+}
+
+impl<'a> InternalEvent for GelfSerializeFailedMissingField<'a> {
```

Comment:
(thinking out loud) I don't know if it's easily accessible, but these could be more generic if the encoder type was passed in as a tag - rather than the as the struct name

Reply:
I like the idea. Upon inspecting the Serializer module and thinking about it, I think the most straight forward way to accomplish that would just to pass it in as a string. Initially I thought, maybe we could put it at the Serializer level in mod.rs , but I don't think that would add much benefit because these internal events will be issued by the specific encoders themselves. So perhaps just defining a static string in the specific encoder file, and passing that in as a tag to the internal event construction would suffice?

Something like:

```Rust
static ENCODER_TYPE_GELF_STR : &str = "GELF";
```
...
```Rust
emit(SerializerFailedMissingField { field, message: MISSING_FIELD_STR, encoder_type: ENCODER_TYPE_GELF_STR });
```

Reply:
Maybe here is an option as well? https://github.com/vectordotdev/vector/blob/master/src/internal_events/prelude.rs